src/docs/build-posix.dox

```diff
@@ -102,7 +103,7 @@ Configure WiredTiger to sleep and wait for a debugger to attach on failure.
 <b>DO NOT</b> configure this option in production environments.
 
 @par \c -DHAVE_DIAGNOSTIC=1
-Configure WiredTiger to perform various run-time diagnostic tests.
+Configure WiredTiger to perform various run-time diagnostic tests (enabled by default).
```

Comment:
Though not related to this PR, it does not feels super safe having the default on for diagnostics. Users need to consciously remember setting `-DHAVE_DIAGNOSTIC=0` for their production builds.

Reply:
Cleaned it up, we enable it by default for non Release build types.